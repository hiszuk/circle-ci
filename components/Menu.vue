<template>
  <div class="menu">
    <MenuItem class="menu-item"
      v-for="item in items"
      :key = "item.label"
      v-bind = "item"
      @clickMenuItem = "onClickMenuItem"
    />
  </div>
</template>

<script>
import MenuItem from '~/components/MenuItem'
export default {
  name: 'Menu',
  components: {
    MenuItem
  },
  props: {
    items: {
      type: Array,
      default: [],
      required: true
    }
  },
  methods: {
    onClickMenuItem({ name }) {
      this.$emit('clickMenuItem', { name })
    },
  },
}
</script>