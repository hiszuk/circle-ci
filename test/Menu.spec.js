import { mount } from "@vue/test-utils";
import Menu from "@/components/Menu.vue";
import menuItems from "@/test/_mockData/menuItems.json";

const factory = () => {
  return mount(Menu, {
    propsData: {
      items: menuItems
    }
  });
};

describe("Menu.vue", () => {
  describe("プロパティ受け渡し", () => {
    it("メニューItem仕様チェック", () => {
      const wrapper = factory();
      expect(wrapper.vm.$props.items[0].label).toBe(menuItems[0].label);
      expect(wrapper.vm.$props.items[0].name).toBe(menuItems[0].name);
      expect(wrapper.vm.$props.items[1].label).toBe(menuItems[1].label);
      expect(wrapper.vm.$props.items[1].name).toBe(menuItems[1].name);
      expect(typeof wrapper.vm.$props.items).toBe("object");
      expect(wrapper.vm.$options.props.items.required).toBe(true);
    });
  });

  describe("template", () => {
    it("snapshot", () => {
      const wrapper = factory();
      // レンダリング結果がスナップショットと同じか
      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });

  describe("MenuClick", () => {
    it("ボタンクリックでメソッドが起動されるか?", () => {
      const wrapper = factory();

      // MENUをクリックする
      wrapper.findAll(".menu-item-label").at(0).trigger("click")
      expect(wrapper.emitted("clickMenuItem")).toBeTruthy();
      expect(wrapper.emitted("clickMenuItem")[0][0]).toEqual({
        name: menuItems[0].name
      });

      // MENUをクリックする
      wrapper.findAll(".menu-item-label").at(1).trigger("click")
      expect(wrapper.emitted("clickMenuItem")).toBeTruthy();
      expect(wrapper.emitted("clickMenuItem")[1][0]).toEqual({
        name: menuItems[1].name
      });
    });
  });

});
