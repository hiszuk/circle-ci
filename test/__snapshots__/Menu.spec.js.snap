// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu.vue template snapshot 1`] = `
<div
  class="menu"
>
  <div
    class="menu-item menu-item"
  >
    <a
      class="menu-item-label"
    >
      
    About
  
    </a>
  </div>
  <div
    class="menu-item menu-item"
  >
    <a
      class="menu-item-label"
    >
      
    Work
  
    </a>
  </div>
</div>
`;
